packages/core/contracts/cross-chain-oracle/OracleSpoke.sol

```diff
@@ -163,29 +163,23 @@ contract OracleSpoke is
     /**
      * @notice Generates stamped ancillary data in the format that it would be used in the case of a price request.
      * @param ancillaryData ancillary data of the price being requested.
-     * @param requester sender of the initial price request.
      * @return the stamped ancillary bytes.
      */
-    function stampAncillaryData(bytes memory ancillaryData, address requester)
-        public
-        view
-        nonReentrantView()
-        returns (bytes memory)
-    {
-        return _stampAncillaryData(ancillaryData, requester);
+    function stampAncillaryData(bytes memory ancillaryData) public view nonReentrantView() returns (bytes memory) {
+        return _stampAncillaryData(ancillaryData);
     }
 
     /**
      * @dev We don't handle specifically the case where `ancillaryData` is not already readily translatable in utf8.
      * For those cases, we assume that the client will be able to strip out the utf8-translatable part of the
      * ancillary data that this contract stamps.
      */
-    function _stampAncillaryData(bytes memory ancillaryData, address requester) internal view returns (bytes memory) {
-        // This contract should stamp its requester's address and network in the
-        // ancillary data so voters can conveniently track the requests path to the DVM.
+    function _stampAncillaryData(bytes memory ancillaryData) internal view returns (bytes memory) {
+        // This contract should stamp the child network's ID and its address so that voters on the parent network can
+        // deterministically track unique price requests back to this contract.
         return
             AncillaryData.appendKeyValueUint(
-                AncillaryData.appendKeyValueAddress(ancillaryData, "childRequester", requester),
+                AncillaryData.appendKeyValueAddress(ancillaryData, "oracleSpoke", address(this)),
```

Comment:
excellent! I really like the idea of including this. very useful in the case we want to upgrade like you mentioned.

There is one downside to this which we should consider: if we _were_ to upgrade, you would not be able to bridge prices that were destined for the previous `OracleSpoke` over the bridge to "seed" the new spoke. infact you would be _forced_ to start with a blank slate with no pricing info and all previous prices are effectively lost.

if we exclude this `childRequester` logic, then we can enable arbitrary archeological prices (requested for and consumed by previous `OracleSpokes`) to be re-used.

another solution to this, other than removing this from the ancillary data, would be to enable chained lookbacks in the same way the Voting.sol contract enables previous Voting instances to be queried to enable data preservation in upgrades.

this, however, might not be a problem at the end of the day as most applications use price once then not again (i.e the expiration of an LSP is once off)

wdyt?

Reply:
Would this be fixed by us either 1) rebridging (if that's possible, depending on the outcome of the other PR) or 2) upgrading the hub oracle at the same time, which would allow any necessary price requests to be rebridged due to _both sides_ being a blank slate.

Thoughts?

Reply:
> There is one downside to this which we should consider: if we were to upgrade, you would not be able to bridge prices that were destined for the previous OracleSpoke over the bridge to "seed" the new spoke. infact you would be forced to start with a blank slate with no pricing info and all previous prices are effectively lost.

Great point @chrismaree 

I think this could be fixed by rebridging but that's suboptimal as we'd like to not have extra transactions. What about just removing the `oracleSpoke` address? Does just the `childChainId` suffice for ancillary price request information?

Reply:
but we could not just re-bridge in the current implementation, no? the `oracleSpoke` address will now differ and so we'll not be able match the ancillary data.

if you remove the `oracleSpoke` then re-bridging becomes posible as the `childChainId` matches. I think this is probs the best solution. wdyt? any downsides with removing this?

Reply:
I agree and I don't think there are any downsides, let's remove `oracleSpoke` so that upgrades are a bit easier.

Reply:
Agree that adding this to the ancillary data isn't strictly necessary and removing it is worthwhile if it gets around all these complications.